apps/src/common.rs

```diff
@@ -1114,18 +1125,35 @@ impl Http3Conn {
 
 impl HttpConn for Http3Conn {
     fn send_requests(
-        &mut self, conn: &mut quiche::Connection, target_path: &Option<String>,
+        &mut self, conn: &mut quiche::Connection, client_args: &ClientArgs,
     ) {
+        let ClientArgs {
+            dump_response_path: target_path,
+            reqs_concurrent,
+            ..
+        } = client_args;
+
+        let max_streams_bidi = conn.get_peer_max_streams_bidirectional();
```

Comment:
Due to the unique way that QUIC does things, this number doesn't match your expectations. Instead peer_streams_left_bidi will let you know how many more streams you _could_ open. And so you'll want to use that number in combination with the number of requests you want to make (not the number you have made) when deciding later to break or not.